src/actions/keyBasedAssetTransferActions.js

```diff
@@ -0,0 +1,367 @@
+        }
+        return keyBasedAssetToTransfer;
+      }),
+    );
+
+    // submit new in queue if no pending
+    const hasPending = keyBasedAssetsToTransferUpdated.some(({ status }) => status === TX_PENDING_STATUS);
+    const transferTransactionsInQueue = keyBasedAssetsToTransferUpdated.filter(({ status }) => !status);
+    if (!hasPending) {
+      if (!isEmpty(transferTransactionsInQueue)) {
+        // submit first in queue
+        const assetToTransferTransaction = transferTransactionsInQueue[0].signedTransaction;
+        const transactionSent = await transferSigned(assetToTransferTransaction?.signedHash || '')
+          .catch((error) => ({ error }));
+        if (!transactionSent?.hash || transactionSent.error) {
+          reportLog('Failed to send key based asset transger signed transaction', {
+            signedTransaction: assetToTransferTransaction,
+            error: transactionSent.error,
+          });
+          return;
+        }
+
+        // update with pending status
+        const updatedTransaction: KeyBasedAssetTransfer = {
+          ...transferTransactionsInQueue[0],
+          status: TX_PENDING_STATUS,
+          transactionHash: transactionSent.hash,
+        };
+        const updatedTransactionSignedHash = updatedTransaction?.signedTransaction?.signedHash;
+        keyBasedAssetsToTransferUpdated = keyBasedAssetsToTransferUpdated
+          .filter(({ signedTransaction }) => signedTransaction?.signedHash === updatedTransactionSignedHash)
+          .concat(updatedTransaction);
+      } else if (!isEmpty(keyBasedAssetsToTransferUpdated)) {
+        // transfer done, reset
+        keyBasedAssetsToTransferUpdated = [];
+        Toast.show({
+          message: 'Key based assets transfer complete!',
+          type: 'success',
+          title: 'Success',
+        });
+      }
+    }
+
+    dispatch(setAndStoreKeyBasedAssetsToTransferAction(keyBasedAssetsToTransferUpdated));
+  };
+};
+
+export const createKeyBasedAssetsToTransferTransactionsAction = (wallet: Wallet) => {
+  return async (dispatch: Dispatch, getState: GetState) => {
+    const {
+      wallet: { data: { address: keyBasedWalletAddress } },
+      accounts: { data: accounts },
+      keyBasedAssetTransfer: { data: keyBasedAssetsToTransfer },
+    } = getState();
+
+    const firstSmartAccount = findFirstSmartAccount(accounts);
+    if (!firstSmartAccount) {
+      reportLog('Failed to find smart wallet account in key based asset transfer creation.');
+      return;
+    }
+
+    /**
+     * we need this to be sequential and wait for each to complete because of local nonce increment
+     * side note: eslint ignored because of "async for" not allowed, however, we need it for sequential calls
+     */
+    const keyBasedAssetsToTransferUpdated = [];
+    for (const keyBasedAssetTransfer of keyBasedAssetsToTransfer) { // eslint-disable-line
+      const signedTransaction = await signKeyBasedAssetTransferTransaction( // eslint-disable-line
```

Comment:
What do you mean?

Reply:
The sync loop - and eslint not liking it 😂

Reply:
Resolving this.

Reply:
It's because the `for..of` is based on generators and is a heavy operation for React Native. I'm using that for..of now on the backend and it makes the code to look so much cleaner! :P